projects/koin-ksp-compiler/src/jvmMain/kotlin/org/koin/compiler/BuilderProcessor.kt

```diff
@@ -67,11 +67,17 @@ class BuilderProcessor(
     }
 
     //TODO Use Koin 4.0 ViewModel DSL
+    @Deprecated("use isKoinComposeViewModelActive")
     private fun isComposeViewModelActive(): Boolean {
-        logger.warn("Use Compose ViewModel for @KoinViewModel generation")
+        logger.warn("[Deprecated] Please use KOIN_USE_COMPOSE_VIEWMODEL arg")
```

Comment:
Why isn't this behind a check if the argument is actually set? I don't even use Compose and this deprecation notice is still printed.

Reply:
is it still present?

Reply:
As of 1.4.0-RC4, yes. Even when explicitly disabling `USE_COMPOSE_VIEWMODEL`.
![the warnings](https://github.com/user-attachments/assets/bb6de1c9-b59d-4c4c-9354-cac7d2e35138)
![ksp config](https://github.com/user-attachments/assets/5e2fc113-0dac-4f0b-af26-b1638ef54f0f)

Looking at the code, it appears the `isComposeViewModelActive` is called in any case, leading to the message being logged without any further checks.
https://github.com/InsertKoinIO/koin-annotations/blob/1.4.0-RC4/projects/koin-ksp-compiler/src/jvmMain/kotlin/org/koin/compiler/BuilderProcessor.kt#L83-L87